container/network.go

```diff
@@ -50,45 +50,67 @@ func NewDockerNetwork() (*DockerNetwork, error) {
 		return nil, err
 	}
 
-	for i := SecondOctet; i < 256; i++ {
-		// IP xxx.xxx.0.1 is reserved for docker network gateway
-		ipv4Addr, ipv4Net, err := net.ParseCIDR(fmt.Sprintf("%d.%d.0.1/16", FirstOctet, i))
-		networkName := fmt.Sprintf("%s_%d_%d", NetworkName, FirstOctet, i)
-		if err != nil {
-			return nil, err
-		}
-		network := &DockerNetwork{
-			client:  c,
-			name:    networkName,
-			ipv4Net: ipv4Net,
-			ipIndex: ipv4Addr,
-		}
-		if err = network.create(); err != nil {
-			fmt.Printf("Failed to create network and retry, err:%v\n", err)
-		} else {
-			return network, nil
-		}
+	network := &DockerNetwork{
+		client:  c,
+		usedIPs: make(map[string]bool, 0),
 	}
 
-	return nil, err
+	if err := network.create(); err != nil {
+		return nil, err
+	}
+
+	if err := network.init(); err != nil {
+		return nil, err
+	}
+
+	return network, nil
 }
 
-// create creates a docker network with given subnet
+// create creates a user-defined docker network
 func (n *DockerNetwork) create() error {
-	ipamConfig := network.IPAMConfig{
-		Subnet: n.ipv4Net.String(),
+	n.name = fmt.Sprintf("%s%d", networkNamePrefix, time.Now().Unix())
+	r, err := n.client.NetworkCreate(context.Background(), n.name, types.NetworkCreate{})
+	if err != nil {
+		return err
 	}
-	ipam := &network.IPAM{
-		Config: []network.IPAMConfig{ipamConfig},
+	n.id = r.ID
+	return nil
+}
+
+func (n *DockerNetwork) init() error {
+	res, err := n.client.NetworkInspect(context.Background(), n.name, types.NetworkInspectOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(res.IPAM.Config) == 0 {
+		return errors.New("Invalid network config.")
 	}
 
-	r, err := n.client.NetworkCreate(context.Background(), n.name, types.NetworkCreate{
-		IPAM: ipam,
-	})
+	ip, ipv4Net, err := net.ParseCIDR(res.IPAM.Config[0].Subnet)
```

Comment:
n.ipIndex, n.ipv4Net, err := net.ParseCIDR(res.IPAM.Config[0].Subnet)

Reply:
fixed in new version.